docs/modules/ROOT/pages/explanation/decisions/portal-api.adoc

```diff
@@ -0,0 +1,78 @@
+= {product} Portal API
+
+== Problem
+
+For the xref:explanation/system/details-ui.adoc[{product} Portal] an API endpoint is needed which implements and exposes the backend functionality.
+This API endpoint is described in xref:explanation/system/details-external-services.adoc[External Services] in more detail.
+As documented there, this API endpoint connects various third-party backend systems together and presents them in a well-defined uniform way for the Portal UI or other API clients to be consumed.
+
+It must be possible to connect to various different backend systems which implement the same functionality so that it is a pluggable system and can be re-used by other parties, which might have different backends for the same task.
+
+{product} API only specifies the interface, the implementation is then specific to the provider of the API.
+
+== Relevant requirements
+
+* xref:references/quality-requirements/maintainability/portal-api-modularity.adoc[Portal API Modularity]
+* xref:references/quality-requirements/reliability/portal-api-data-recoverability.adoc[Portal API Data Recoverability]
+* xref:references/quality-requirements/performance/portal-api-data-freshness.adoc[Portal API Data Freshness]
+
+== Proposals
+
+[qanda]
+Custom API implementation::
+Implementation of a custom (REST or GraphQL) API which connects to all the backend system and provides the abstraction to them.
+This can be a synchronous or asynchronous connection, depending on the implementation.
+Authentication, authorization and other core API features will have to be implemented specific to this API.
+
+Kubernetes API::
+Reuse the Kubernetes API to leverage all the core-plumbing which already exists: Authentication, RBAC, scalability, well-defined resource-model, extensibility with `CRDs`, aggregated API server and much more.
+This seems to be a common topic in the Cloud Native world, see for example the experiment called https://github.com/kcp-dev/kcp[kcp - a minimal Kubernetes API server].
+
+== Decision
+
+Kubernetes API
+
+== Rationale
+
+The Kubernetes API server is a perfect match for this use-case. With it, we can many core API plumbing features "for free":
+
+Authentication::
+We can leverage what we already use for the {zone}s.
+Authentication is built-in to the Kubernetes API server and it's even pluggable.
+
+Authorization::
+We can make use of standard Kubernetes RBAC access control and could even enhance it with a policy engine like Kyverno.
+
+Asynchronous::
+It allows us to build upon an asynchronous architecture which when working with multiple backend systems will prove to be more stable and easier to implement.
+
+Persistent Storage::
+Storing data is built-in directly in the core.
+We can store temporary data without having to implement a storage backend.
+
+API Extensibility::
+The API can easily be extended by leveraging `CustomResourceDefinition` (CRDs).
```

Comment:
This is not really a reason that favors one over the other. I mean, REST APIs can also be extended. I would either remove it or move to the proposal

Reply:
While it's true that REST APIs can also be extended, it's so much simpler to do with the Kubernetes API that I'd like to keep it here to emphasize this. Extensibility is a core feature of the Kubernetes API.

Reply:
Learning from the K8up v2 case (with renamings, adding and removing fields etc.) we both know that this extensibility is not so easy in practice because CRDs are both API specs AND data/document record at the same time.
With that argument, I'd say that a REST API approach has the advantage that it doesn't need to deal with persistence since it's abstracted away. The extensibility from Kubernetes is more limited because of the data records being directly attached to spec.

Reply:
That's not really true, the Kubernetes API would offer a lot of tooling to deal with changing API specs and converting them between API versions. The API versioning plumbing the Kubernetes API brings is very well-thought-out. What we did in K8up v2 is to actually rename the whole API, not just changing spec or version.

Reply:
Well, still. In case of data scheme changes (the spec in CRD) that change needs to be populated into the clients as well (e.g. frontend, CI/CD systems etc.). With both approaches, only non-breaking extensions can be transparently added.
So, considering that in both approaches breaking changes need changes in all clients, I think this argument is basically redundant and I would simply remove it :shrug: 

What is the exact reason why you think extensions in Kubernetes is "simpler" to do than with REST?

Reply:
Because the Kubernetes API is already prepared and _a lot of thought_ has been put into all the concepts of extensibility. As well there is a lot of plumbing around to assist with changing API specs and gradually moving to breaking changes of the API.

This could all be done with a custom API as well, but why should we want to re-invent the wheel of this "boring" stuff again when others already solved it for us so that we can focus on the actual case?